internal/pkg/template/template.go

```diff
@@ -63,6 +99,49 @@ func (t *Template) Parse(path string, data interface{}, options ...ParseOption)
 	return &Content{buf}, nil
 }
 
+// UploadEnvironmentCustomResources uploads the environment custom resource scripts.
+func (t *Template) UploadEnvironmentCustomResources(upload func(string, ...Uploadable) (string, error)) ([]string, error) {
```

Comment:
nit: okay to remove the weird slice type conversion in `env init`, maybe we should just do:
```go
// UploadFn is invoked to zip multiple template contents and upload them to an S3 bucket under the specified key.
type UploadFunc(key string, objects ...s3.NamedBinary) (urls []string, err error)
```
An example of this in the standard library is the `WalkDirFunc`: https://golang.org/pkg/path/filepath/#WalkDir

and this method becomes:
```go
func (t *Template) UploadEnvironmentCustomResources(fn UploadFunc) ([]string, error) {
```
which allows `Uploadable` to become private: `uploadableContent`

Reply:
According to the std lib example above, we would define `UploadFunc` in the `s3` pkg